pkg/apiserver/rest/usecase/config.go

```diff
@@ -134,12 +135,27 @@ func (u *configUseCaseImpl) GetConfigType(ctx context.Context, configType string
 }
 
 func (u *configUseCaseImpl) CreateConfig(ctx context.Context, req apis.CreateConfigRequest) error {
+	p := req.Properties
+	// If the component is Terraform type, set the provider name same as the application name and the component name
+	if strings.HasPrefix(req.ComponentType, types.TerrfaormComponentPrefix) {
```

Comment:
we should handle it in the cue script. such as:

```
...
provider_name: context.name
...
```

there is no need to add a field named `name`.

Reply:
For cli users, it's used to indicate the names of the provider, and it has the default name for a specific provider like https://github.com/oam-dev/catalog/blob/master/addons/terraform-baidu/definitions/baidu-provider.cue#L64. `context.name` could not express the default value.

Reply:
And the default provider name is to be compatible with previous addons, within which an provider will be created.

Reply:
@zzxwill For the CLI users, what is the use case like?

Reply:
If he must create a component, it must have a component name. What is the starting point for making the component name and provider name inconsistent here?

Reply:
For cli users, we create a provider whose name is decided by `--name`.
```
➜  kubevela git:(terraform-ui) ✗ vela provider add terraform-alibaba -h
Authenticate Terraform Cloud Provider terraform-alibaba by creating a credential secret and a Terraform Controller Provider

Usage:
  vela provider add terraform-alibaba [flags]

Examples:
vela provider add terraform-alibaba

Flags:
      --ALICLOUD_ACCESS_KEY string   Get ALICLOUD_ACCESS_KEY per this guide https://help.aliyun.com/knowledge_detail/38738.html
      --ALICLOUD_REGION string       Get ALICLOUD_REGION by picking one RegionId from Alibaba Cloud region list https://www.alibabacloud.com/help/doc-detail/72379.htm
      --ALICLOUD_SECRET_KEY string   Get ALICLOUD_SECRET_KEY per this guide https://help.aliyun.com/knowledge_detail/38738.html
  -h, --help                         help for terraform-alibaba
      --name default                 The name of Terraform Provider for Alibaba Cloud, default is default (default "default")

Global Flags:
  -y, --yes   Assume yes for all user prompts
```

There won't be any inconsistencies as the default name is the provider name, like https://github.com/oam-dev/catalog/blob/master/addons/terraform-baidu/definitions/baidu-provider.cue#L21.

Reply:
@zzxwill So, both the CLI and the API will create an application that includes a component. you could set the component name by value from `--name`. no need set a parameter with the name `name`.

For the default value:

* API: no need set it, users must set a name for the configuration.
* CLI: we could set a default value for the different provider kinds.

Reply:
@barnettZQG Correct! For CLI users, we can set the default value for difficult providers.
```
switch {
  case "terraform-alibaba":
    name := "default"
  case "terraform-aws":
    name := "aws"
...
}
```
In this case, the implementation for VelaUX is simpler, and the parameter `name` in the definition is not needed. In the original case, the implementation for CLI is simpler.

Reply:
About the default name, is there any other reason besides compatibility?

Reply:
default name == provider name?

Reply:
@barnettZQG 

> About the default name, is there any other reason besides compatibility?

No. All definitions also uses the default name in the mainifests.

Reply:
> default name == provider name?

Correct.

Reply:
Define the UI schemas for all providers, and hide the named field in UI.